integrations/weaviate/src/haystack_integrations/document_stores/weaviate/document_store.py

```diff
@@ -172,11 +172,15 @@ def client(self):
         if self._client:
             return self._client
 
-        if self._url and self._url.startswith("http") and self._url.endswith(".weaviate.network"):
+        if (
```

Comment:
I wonder under what circumstances we could now reach the else branch. We wouldn't reach the else branch for example if we have `https://grpc-....c0.us-west3.gcp.weaviate.cloud` as url. It's not well explained when we would go to the else branch and we should improve that later.

Reply:
The `if` branch works for Weaviate cloud.
The `else` branch works for embedded (no `self._url`) or local docker (URL does not end with .weaviate.network or .weaviate.cloud).

This can be informative: https://github.com/weaviate/weaviate-python-client/blob/main/weaviate/connect/helpers.py

Thinking more on this, my impression is that `self._url.startswith("http")` should be removed from the condition.
This is a perfectly fine URL: `"rAnD0mD1g1t5.something.weaviate.cloud"` and should be allowed.

Reply:
Okay, thanks, makes sense. Let's remove `self._url.startswith("http")` too and then I'll approve.

Reply:
You could use `if self._url and self._url.endswith((".weaviate.network", ".weaviate.cloud")):`

Reply:
thanks.Done